fastapi_jsonapi/data_layers/filtering/sqlalchemy.py

```diff
@@ -78,19 +95,104 @@ def create_filter(self, schema_field: ModelField, model_column, operator, value)
         types = [i.type_ for i in fields]
         clear_value = None
         errors: List[str] = []
-        for i_type in types:
-            try:
-                if isinstance(value, list):  # noqa: SIM108
-                    clear_value = [i_type(item) for item in value]
-                else:
-                    clear_value = i_type(value)
-            except (TypeError, ValueError) as ex:
-                errors.append(str(ex))
+
+        pydantic_types, userspace_types = self._separate_types(types)
+
+        if pydantic_types:
+            if isinstance(value, list):
+                clear_value, errors = self._cast_iterable_with_pydantic(pydantic_types, value)
+            else:
+                clear_value, errors = self._cast_value_with_pydantic(pydantic_types, value)
+
+        if clear_value is None and userspace_types:
+            log.warning("Filtering by user type values is not properly tested yet. Use this on your own risk.")
+
+            cast_failed = object()
+            clear_value = cast_failed
+
+            for i_type in types:
+                try:
+                    if isinstance(value, list):  # noqa: SIM108
+                        clear_value = [i_type(item) for item in value]
+                    else:
+                        clear_value = i_type(value)
+                except (TypeError, ValueError) as ex:
+                    errors.append(str(ex))
+
+            if clear_value is cast_failed:
+                raise InvalidType(
+                    detail=f"Can't cast filter value `{value}` to user type.",
```

Comment:
```suggestion
                    detail=f"Can't cast filter value `{value}` to arbitrary type.",
```

предлагаю убрать `user`, так как может ввести в заблуждение